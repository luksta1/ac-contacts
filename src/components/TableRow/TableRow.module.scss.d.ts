export const block: string;
export const cell: string;
export const cellContact: string;