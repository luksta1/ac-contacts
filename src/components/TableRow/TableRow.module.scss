@import "../../styles/variables/colors";

.block {
    border-top: 2px solid $ac-slate-7;
    height: 2.875em;
    width: 100%;

    &:nth-child(even) {
        background: $ac-slate-8;
    }

    @media only screen 
        and (max-width: 960px) {
            border-left: 2px solid $ac-slate-7;
            border-top: none;
            height: 100%;
        }
}

.cell {
    padding: 0 2em;

    @media only screen 
        and (max-width: 960px) {
            align-items: center;
            display: flex;
            height: calc((100% -8em) / 5);
        }

    h2 {
        color: $ac-slate-1;
        font-size: 0.938em;
        font-weight: default;
    }

    &Contact {
        align-items: center;
        display: flex;

        img {
            width: 1.5em;
            height: 1.5em;
            border-radius: 50%;
            margin-right: 0.8em;
        }

        h2 {
            color: $ac-ocean-1;
            font-weight: 500;
        }
    }

    &Deals {

        h2 {
            margin-left: 1em;

            @media only screen 
            and (max-width: 960px) {
                margin-left: 0;
            }
        }
    }

    &Tags {
        white-space: nowrap;
        max-width: 25%;

        h2 {
            max-width: 25%;
            text-overflow: ellipsis;
        }
    }
}

