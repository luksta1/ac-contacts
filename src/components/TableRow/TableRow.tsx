import React from 'react';

const TableRow: React.FC = () => {
  return (
    <></>
  );
}

export default TableRow;
