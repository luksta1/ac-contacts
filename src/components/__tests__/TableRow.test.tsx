import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import TableRow from '../TableRow';


describe('TableHeader component', () => {
    const props = {
        contact: {
            contact: 'Test Name',
            deals: 2,
            id: '27',
            location: {
                city: 'Chicago',
                state: 'Illinois',
                country: 'US',
            },
            tags: ['tag'],
            totalValue: 11200
        },
        columns: ['col1', 'col2', 'col3']
    }

    it('renders without crashing', () => {
        const tBody = document.createElement('tbody');
        ReactDOM.render(<TableRow {...props} />, tBody);
        ReactDOM.unmountComponentAtNode(tBody);
    });

    let component: any;

    beforeEach(() => {
        component = shallow(
            <TableRow {...props} />
        );
    })

    it('matches the snapshot', () => {
        expect(component).toMatchSnapshot();
    });

    it('has renders a <tr> element', () => {
        expect(component.find('tr').length).toEqual(1);
    });

    it('has renders a <td> element for each column', () => {
        expect(component.find('td').length).toEqual(3);
    });
})