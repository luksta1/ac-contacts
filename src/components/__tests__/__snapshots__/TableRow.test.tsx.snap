// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeader component matches the snapshot 1`] = `ShallowWrapper {}`;
